src/compiler/checker.ts

```diff
@@ -382,10 +382,15 @@ namespace ts {
 
         /** Returns true if node1 is defined before node 2**/
         function isDefinedBefore(node1: Node, node2: Node): boolean {
+            if (isAliasSymbolDeclaration(node1) && !isInternalModuleImportEqualsDeclaration(node1)) {

```

Comment:
I feel like if they are external aliases, that doesn't mean node1 is before node2, it means node1 doesn't have to be before node2. It feels strange to return true for isDefinedBefore in this case.